arch/arm/src/nrf52/nrf52_gpio.c

```diff
@@ -192,6 +202,59 @@ static inline void nrf52_gpio_sense(nrf52_pinset_t cfgset,
   putreg32(regval, offset);
 }
 
+/****************************************************************************
+ * Name: nrf52_gpio_drive
+ *
+ * Description:
+ *   Set DRIVE configuration for a pin
+ *
+ ****************************************************************************/
+
+static inline void nrf52_gpio_drive(nrf52_pinset_t cfgset,
```

Comment:
This needs to take the lock or use `modifyreg32` which does it for you.  I notice that a lot of the functions here to do not do this and should probably also be updated.

Reply:
since this would then require changing various other parts of this code do you mind if I do it in a separate PR?

Reply:
Sure, its probably fine to just take the lock on the public functions that need it since there is not waiting going on in the private ones.  It looks like that is only `nrf52_gpio_config` so very minor change just calling `spin_lock_irqsave` at the top.  Do you want to do that change or should I?

Reply:
ok, I followed your suggestion, let me know if it looks OK

Reply:
Looks good.  Thanks!